src/PowerShellEditorServices/Language/FindDeclarationVisitor.cs

```diff
@@ -76,19 +78,68 @@ public override AstVisitAction VisitFunctionDefinition(FunctionDefinitionAst fun
+            {
+                return AstVisitAction.Continue;
+            }
+
+            // The AssignmentStatementAst could contain either of the following Ast types:
+            // VariableExpressionAst, ArrayLiteralAst, ConvertExpressionAst, AttributedExpressionAst
+            // We might need to recurse down the tree to find the VariableExpressionAst we're looking for
+            List<VariableExpressionAst> asts = FindMatchingAsts(assignmentStatementAst.Left, variableName);
+            if (asts.Count > 0)
+            {
+                FoundDeclaration = new SymbolReference(SymbolType.Variable, asts.FirstOrDefault().Extent);
+                return AstVisitAction.StopVisit;
+            }
+            return AstVisitAction.Continue;
+        }
+
+        /// <summary>
+        /// Takes in an ExpressionAst and recurses until it finds all VariableExpressionAst and returns the list of them.
+        /// We expect this ExpressionAst to be either: VariableExpressionAst, ArrayLiteralAst, ConvertExpressionAst, AttributedExpressionAst
+        /// </summary>
+        /// <param name="ast">An ExpressionAst</param>
+        /// <param name="variableName">The name of the variable we are trying to find</param>
+        /// <returns>A list of ExpressionAsts that match the variable name provided</returns>
+        private static List<VariableExpressionAst> FindMatchingAsts (ExpressionAst ast, string variableName) {
+
+            // VaraibleExpressionAst case - aka base case. This will return a list with the variableExpressionAst in it or an empty list if it's not the right one.
+            var variableExprAst = ast as VariableExpressionAst;
+            if (variableExprAst != null)
+            {
+                if (variableExprAst.VariablePath.UserPath.Equals(
                     variableName,
                     StringComparison.OrdinalIgnoreCase))
+                {
+                    return new List<VariableExpressionAst>{ variableExprAst };
+                }
+                return new List<VariableExpressionAst>();
+            }
+
+            // VariableExpressionAsts could be an element of an ArrayLiteralAst. This adds all the elements to the call stack.
+            var arrayLiteralAst = ast as ArrayLiteralAst;
+            if (arrayLiteralAst != null)
             {
-                return AstVisitAction.Continue;
+                return (arrayLiteralAst.Elements.SelectMany(e => FindMatchingAsts(e, variableName)).ToList());
             }
 
-            // TODO also find instances of set-variable
-            FoundDeclaration = new SymbolReference(SymbolType.Variable, variableExprAst.Extent);
-            return AstVisitAction.StopVisit;
+            // The ConvertExpressionAst (static casting for example `[string]$foo = "asdf"`) could contain a VariableExpressionAst so we recurse down
+            var convertExprAst = ast as ConvertExpressionAst;
+            if (convertExprAst != null && convertExprAst.Child != null)
```

Comment:
Ideally we're moving to the switch, but the null check here could be compressed to `convertExprAst?.Child != null`

Reply:
Oh but apparently this works

```powershell
($a, $b) = 1, 2
```

So replace it with `ParenExpressionAst`  :P